model/buildlogger_storage.go

```diff
@@ -27,18 +27,35 @@ func (lf LogFormat) Validate() error {
 	}
 }
 
-// LogArtifact is a type that describes a sub-bucket of logs stored in s3. It
-// is the bridge between S3-based offline log storage and the cedar-based log
-// metadata storage. The prefix field indicates the name of the sub-bucket. The
-// top level bucket is accesible via the cedar.Environment interface.
+// LogArtifact describes a sub-bucket of logs stored in s3. It is the bridge
+// between S3-based offline log storage and the cedar-based log metadata
+// storage. The prefix field indicates the name of the sub-bucket. The top
+// level bucket is accesible via the cedar.Environment interface.
 type LogArtifactInfo struct {
 	Prefix      string             `bson:"prefix"`
 	Permissions pail.S3Permissions `bson:"permissions"`
 	Version     int                `bson:"version"`
+	Chunks      []LogChunkInfo     `bson:"chunks"`
 }
 
 var (
 	logArtifactInfoPrefixKey      = bsonutil.MustHaveTag(LogArtifactInfo{}, "Prefix")
 	logArtifactInfoPermissionsKey = bsonutil.MustHaveTag(LogArtifactInfo{}, "Permissions")
 	logArtifactInfoVersionKey     = bsonutil.MustHaveTag(LogArtifactInfo{}, "Version")
+	logArtifactInfoChunksKey      = bsonutil.MustHaveTag(LogArtifactInfo{}, "Chunks")
+)
+
+// LogChunkInfo describes a chunk of log lines stored in s3.
+type LogChunkInfo struct {
+	Key      string `bson:"key"`
+	NumLines int    `bson:"num_lines"`
+	Start    int64  `bson:"start"`
+	End      int64  `bson:"end"`
```

Comment:
shouldn't these be either (preferably) time.Time instances?

Reply:
No, the timestamps are stored as milliseconds since the unix epoch so these should be in64s.

Reply:
I think we should convert them in the application/database because anything we can do to make handling time math easier would be good.